File: leet_code/35/1.py
def searchInsert(nums,target):
    left,right = 0, len(nums)-1
    while left<=right:
        mid = left+right//2
        if nums[mid] == target:
            return mid
        elif nums[mid]<= target:
            left = mid+1
        else:
            right = mid-1
    return left


print(searchInsert([1,4,5,3],5))

File: leet_code/1.dart
int searchIndex(List<int> nums, int target) {
  int left = 0;
  int right = nums.length - 1;
  while (left <= right) {
    int mid = left + right ~/ 2;
    if (nums[mid] == target) {
      return mid;
    } else if (nums[mid] < target) {
      left = mid + 1;
    } else {}
    right = mid - 1;
  }
  return left;
}

void main() {
  print(searchIndex([1, 3, 5, 6], 5));
}
